openmrs-19/src/test/java/org/motechproject/openmrs19/tasks/OpenMRSTaskDataProviderTest.java

```diff
@@ -47,6 +52,57 @@ public void shouldReturnNullWhenClassIsNotSupported() {
     }
 
     @Test
+    public void shouldReturnNullWhenWrongLookupNameForEncounter() {
+        String className = OpenMRSEncounter.class.getSimpleName();
+        Object object = taskDataProvider.lookup(className, "wrongLookupName", null);
+
+        assertNull(object);
+        verifyZeroInteractions(encounterService);
+    }
+
+    @Test
+    public void shouldReturnNullWhenEmptyLookupFieldsForLookupGetEncounterByUuid() {
+        String className = OpenMRSEncounter.class.getSimpleName();
+        Map<String, String> lookupFields = new HashMap<>();
+
+        when(encounterService.getEncounterByUuid(null)).thenReturn(null);
+        Object object = taskDataProvider.lookup(className, BY_UUID, lookupFields);
+
+        assertNull(object);
+        verify(encounterService).getEncounterByUuid(null);
+    }
+
+    @Test
+    public void shouldReturnNullWhenEncounterNotFoundForLookupGetEncounterByUuid() {
+        String className = OpenMRSEncounter.class.getSimpleName();
+        Map<String, String> lookupFields = new HashMap<>();
+        lookupFields.put(UUID, "4");
+
+        when(encounterService.getEncounterByUuid("4")).thenReturn(null);
+        Object object = taskDataProvider.lookup(className, BY_UUID, lookupFields);
+
+        assertNull(object);
+        verify(encounterService).getEncounterByUuid("4");
+    }
+
+    @Test
+    public void shouldReturnEncounterForLookupGetEncounterByUuid() {
```

Comment:
Additionally the default behaviour for mock objects is 'return null if when not defined', so in this test when().thenReturn() is redundant, because even if we not defined this the encounterService mock will return null. However in my opinion it is more clear when reading the code.